float32 pwm
string direction
